cardano-cli/src/Cardano/CLI/Byron/Delegation.hs

```diff
@@ -103,10 +103,22 @@ checkDlgCert cert magic issuerVK' delegateVK' =
     epoch :: EpochNumber
     epoch = unAnnotated $ Dlg.aEpoch cert
 
-    cert' :: Dlg.ACertificate ByteString
-    cert' = cert { Dlg.aEpoch = Annotated epoch (serialize' epoch)
-                 , Dlg.annotation = serialize' (void cert')
-                 }
+    cert' :: Dlg.ACertificate ()
+    cert' =
+      Dlg.UnsafeACertificate
+      { Dlg.aEpoch     = Annotated epoch ()
+      , Dlg.issuerVK   = Dlg.issuerVK cert
+      , Dlg.delegateVK = Dlg.delegateVK cert
+      , Dlg.signature  = Dlg.signature cert
+      , Dlg.annotation = ()
+      }
+
+    cert'' :: Dlg.ACertificate ByteString
+    cert'' =
+      cert'
+      { Dlg.aEpoch = Annotated epoch (serialize' epoch)
+      , Dlg.annotation = serialize' $ cert'
+      }
```

Comment:
I would have gone for something like (didn't type check it):
```haskell
cert' =
  let unannotated = cert { Dlg.aEpoch = Annotated epoch (serialize' epoch) }
  in unannotated { Dlg.annotation = serialize' unannotated }
```
(Maybe it needs a `unannotated :: Dlg.Certificate` signature?)